snakemake/workflow.py

```diff
@@ -1197,6 +1198,9 @@ def include(
         basedir = self.current_basedir if self.included_stack else None
         snakefile = infer_source_file(snakefile, basedir)
 
+        if "HOSTING_PROVIDER_TOKEN" in os.environ and isinstance(snakefile, HostingProviderFile):
+            snakefile.set_token(os.environ["HOSTING_PROVIDER_TOKEN"])
```

Comment:
Instead of using set_token here, I'd rather always check for the token in the constructor of GithubFile and GitlabFile. This way, one does not even need to modify the usage of them in the codebase. Also, I would rather use service specific tokens. Usually, people use the env vars `GITHUB_TOKEN` and `GITLAB_TOKEN`.

Reply:
Oh yeah, I didn't realise that this wasn't needed anymore if not fetching an argument from the workflow. That's much better.